hyperspy/io_plugins/hdf5.py

```diff
@@ -339,20 +339,40 @@ def dict2hdfgroup(dictionary, group, compression=None):
             group.attrs["_datetime_" + key] = repr(value)
         elif isinstance(value, list):
             if len(value):
-                dict2hdfgroup(dict(zip(
-                    [unicode(i) for i in xrange(len(value))], value)),
-                    group.create_group(
-                    '_list_' + str(len(value)) + '_' + key),
-                    compression=compression)
+                try:
+                    tmp = np.array(value)
+                except ValueError:
+                    tmp = np.array([[0]])
+                if tmp.dtype is np.dtype('O') or tmp.ndim is not 1:
```

Comment:
For consistency. If no such exception is made, then [(1,2),(3,4)] will get turned into one array of ndim=2, but [(1,2), (3,4,5)] - into a list of two arrays, each of ndim=1. So here I make sure it's always one-dimensional. Effectively preventing numpy from "eating up" not only the attribute types (float, string), but also outer layer type (tuple, list)